core/src/main/java/com/alipay/sofa/common/log/env/LogEnvUtils.java

```diff
@@ -83,39 +91,68 @@ public static boolean isCommonsLoggingUsable(ClassLoader spaceClassloader) {
 
     /**
      * 对通用的全局日志相关的配置变量进行整理
+     * logging.path
+     * loggingRoot
+     * file.encoding
+     * PID
      */
-    public static void processGlobalSystemLogProperties() {
+    public static Map<String, String> processGlobalSystemLogProperties() {
+        if (globalSystemProperties != null) {
+            return globalSystemProperties;
+        }
+
+        Map<String, String> properties = new HashMap<String, String>();
 
         /**
-         *  == global.以系统变量file.encoding为准,获取不到再默认设置为UTF-8
+         * 设置进程号上下文，用于日志打印.
          */
-
-        if (System.getProperty(LOG_ENCODING_PROP_KEY) == null) {
-            System.setProperty(LOG_ENCODING_PROP_KEY, UTF8_STR);
-        }
+        properties.put(PROCESS_MARKER, ProcessIdUtil.getProcessId());
 
         /**
-         *  == global.logging.path
+         *  以系统变量file.encoding为准,获取不到再默认设置为UTF-8
          */
+        properties.put(LOG_ENCODING_PROP_KEY, System.getProperty(LOG_ENCODING_PROP_KEY, UTF8_STR));
 
+        /**
+         *  以系统变量 logging.path 和 loggingRoot 为准，优先 logging.path 配置项
+         */
         String loggingPath = System.getProperty(LOG_PATH);
         String loggingRoot = System.getProperty(OLD_LOG_PATH);
-        //以loggingPath为准（可覆盖loggingRoot）
-        if (loggingPath != null && !loggingPath.equalsIgnoreCase(loggingRoot)) {
-            System.setProperty(OLD_LOG_PATH, loggingPath);
-            return;
+        if (!StringUtil.isBlank(loggingPath)) {
+            //以loggingPath为准（可覆盖loggingRoot）
+            loggingRoot = loggingPath;
+        } else if (!StringUtil.isBlank(loggingRoot)) {
+            // only loggingRoot is configured
+            loggingPath = loggingRoot;
+        } else {
+            //还是提供一个默认值$HOME/logs,否则中间件单元测试报错,需要手动设置路径
+            loggingPath = LOGGING_PATH_DEFAULT;
+            loggingRoot = LOGGING_PATH_DEFAULT;
         }
-        // only loggingRoot is configured
-        if (loggingRoot != null && (loggingPath == null || loggingPath.isEmpty())) {
-            System.setProperty(LOG_PATH, loggingRoot);
+        properties.put(LOG_PATH, loggingPath);
+        properties.put(OLD_LOG_PATH, loggingRoot);
+
+        //设置 logging.level.* 和 logging.path.* 配置
+        for (String key : System.getenv().keySet()) {
```

Comment:
use `entrySet()`

Reply:
it just cares about key， wont fetch the value.

Reply:
`properties.put(key.toLowerCase(), System.getenv().get(key));`